qemu/tests/qemu_guest_agent.py

```diff
@@ -1354,6 +1354,147 @@ def gagent_check_get_disks(self, test, params, env):
                     test.fail("Disk %s alias is defferent "
                               "between guest and qga." % diskname)
 
+    @error_context.context_aware
+    def gagent_check_ssh_public_key_injection(self, test, params, env):
+        """
+        Execute commands related inject ssh_public_keys to guest agent,
+        and all of these following commands need to be checked via another
+        family command "guest-ssh-get-authorized-keys".
+        Steps:
+        1) Check basic function whether can work well.
+        2) check the result of adding new ssh keys that
+         one of them was existed.
+        3) remove ssh keys
+        4) reset ssh keys.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        def ssh_key_test(operation, *keys, **kwargs):
+            """
+            Do ssh_key generation and ssh_login test to confirm
+            whether the ssh_key api work well or not.
+
+            :param operation:
```

Comment:
Should not be empty.

Reply:
sorry, my bad, new code has been updated.